game/ato/flight.py

```diff
@@ -122,6 +122,8 @@ def missing_pilots(self) -> int:
     def return_pilots_and_aircraft(self) -> None:
         self.roster.clear()
         self.squadron.claim_inventory(-self.count)
+        # Also release any pilots reserved by the autoplanner
+        self.squadron.release_autoplanner_pilot_reservations()
```

Comment:
Why do planned flights managing unrelated pilot reservations? Why does returning _one_ flight cancel _all_ reservations in the squadron? This needs a comment explaining why this needs to happen here.